app/grandchallenge/reader_studies/views.py

```diff
@@ -1429,3 +1468,326 @@ def form_valid(self, form):
+        context.update(
+            {
+                "display_set": self.kwargs["pk"],
+                "interface": self.kwargs["interface_pk"],
+                "slug": self.kwargs["slug"],
+            }
+        )
+        return context
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        if self.interface.is_image_kind:
+            linked_task = add_image_to_display_set.signature(
+                kwargs={
+                    "display_set_pk": self.kwargs["pk"],
+                    "interface_pk": self.kwargs["interface_pk"],
+                },
+                immutable=True,
+            )
+            kwargs.update(
+                {
+                    "user": self.request.user,
+                    "linked_task": linked_task,
+                    "auto_id": f"id-{self.kwargs['pk']}-%s",
+                }
+            )
+        else:
+            kwargs.update(
+                {
+                    "user": self.request.user,
+                    "display_set": self.display_set,
+                    "interface": self.interface,
+                    "auto_id": f"id-{self.kwargs['pk']}-%s",
+                }
+            )
+        return kwargs
+
+    def form_valid(self, form):
+        if self.interface.is_image_kind:
+            messages.add_message(
+                self.request, messages.SUCCESS, "Image import started."
+            )
+        return super().form_valid(form)
+
+    def get_success_url(self):
+        return reverse(
+            "reader-studies:display-set-detail",
+            kwargs={"pk": self.kwargs["pk"], "slug": self.kwargs["slug"]},
+        )
+
+
+class DisplaySetInterfacesCreate(ObjectPermissionRequiredMixin, FormView):
+    form_class = DisplaySetInterfacesCreateForm
+    permission_required = (
+        f"{ReaderStudy._meta.app_label}.change_{DisplaySet._meta.model_name}"
+    )
+    raise_exception = True
+
+    def get_permission_object(self):
+        return self.display_set
+
+    @property
+    def display_set(self):
+        if self.kwargs.get("pk"):
+            return DisplaySet.objects.get(pk=self.kwargs["pk"])
+
+    @property
+    def reader_study(self):
+        if self.display_set:
+            return self.display_set.reader_study
+        else:
+            return ReaderStudy.objects.get(slug=self.kwargs["slug"])
+
+    def get_template_names(self):
+        if self.kwargs.get("pk"):
+            return ["reader_studies/display_set_interface_create.html"]
+        else:
+            return ["reader_studies/display_set_new_interface_create.html"]
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update(
+            {
+                "pk": self.kwargs.get("pk"),
+                "reader_study": self.reader_study,
+                "interface": self.request.GET.get("interface"),
+                "user": self.request.user,
+                "auto_id": f"id-{random.randint(0, 10**10)}-%s",
```

Comment:
There is not a structured way to do this?

Reply:
I couldn't think of a better way unfortunately. We need a unique id to use as the htmx target, but there could be more than one of these forms loaded in the create view

Reply:
A UUID?